src/screens/SSICredentialsRequiredScreen/index.tsx

```diff
@@ -50,10 +53,17 @@ const SSICredentialsRequiredScreen: FC<Props> = (props: Props): JSX.Element => {
           restrictToFormats: format,
           restrictToDIDMethods: subjectSyntaxTypesSupported,
         });
-        const matchedVCs: Array<UniqueVerifiableCredential> = selectResult.verifiableCredential
-          ? selectResult.verifiableCredential
-              .map((matchedVC: IVerifiableCredential) => getMatchingUniqueVerifiableCredential(uniqueVCs, matchedVC))
-              .filter((matchedVC): matchedVC is UniqueVerifiableCredential => !!matchedVC) // filter out the undefined (should not happen)
+        if (selectResult.areRequiredCredentialsPresent === "error") {
+          console.info('pex.selectFrom returned errors:\n', JSON.stringify(selectResult.errors));
+        }
+        const matchedVCs: Array<UniqueVerifiableCredential> = selectResult.matches && selectResult.verifiableCredential
+          ? selectResult.matches.map((match: SubmissionRequirementMatch) => {
+            const matchedVC = JSONPath.query(selectResult, match.vc_path[0])
```

Comment:
can we add some comments explaining why we only pick the first one?

Reply:
Can we have multiple vc_path elements for a single match?

Reply:
Not entirely sure from the top of my head   but given the json path would allow for array results it stems to reason that multiple matches should be allowed yes and that all should be returned